File: ros_basics/src/int32_subscriber.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"

#include <iostream>

void callback(const std_msgs::Int32::ConstPtr& msg)
{
	ROS_INFO("Subscriber received: %d", msg->data);
}

int main(int argc, char **argv)
{
	ros::init(argc, argv, "int32_subscriber");
	ros::NodeHandle nh;

	ros::Subscriber sub = nh.subscribe("/integer",10,callback);

	ros::spin();

	return 0;
}

File: ros_basics/scripts/int32_subscriber.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Int32

def callback(data):
	rospy.loginfo("Subscriber heard : %d", data.data)

def subscriber():
	rospy.init_node("int32_subscriber", anonymous =True)
	rospy.Subscriber("integer", Int32, callback)
	rospy.spin()

if __name__ == '__main__':
	subscriber()